homeassistant/components/switch/dli.py

```diff
@@ -0,0 +1,91 @@
+"""
+Support for Digital Loggers Inc Web Power Controllers
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/switch.dli/
+"""
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components.switch import (SwitchDevice, PLATFORM_SCHEMA)
+from homeassistant.const import CONF_URL, CONF_USERNAME, CONF_PASSWORD
+import homeassistant.helpers.config_validation as cv
+
+REQUIREMENTS = ['dlipower==0.7.165']
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_URL): cv.string,
+    vol.Required(CONF_USERNAME): cv.string,
+    vol.Required(CONF_PASSWORD): cv.string
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup the Web Power Controller."""
+    import dlipower
+    switch = dlipower.PowerSwitch(hostname=config.get(CONF_URL),
+                                  userid=config.get(CONF_USERNAME),
+                                  password=config.get(CONF_PASSWORD))
+    switches = []
+    for outlet in switch.statuslist():
+        if outlet[2] == 'ON':
+            state = True
+        else:
+            state = False
+        if outlet[1] is None:
+            switches.append(
+                DLISwitch(
+                    outlet[0],
+                    config.get(CONF_URL) + ':' + str(outlet[0]),
```

Comment:
Why not use string formatting here? `'{}:{}'.format(config.get(CONF_URL), outlet[0])`